src/base/memory_monitor_sr.cpp

```diff
@@ -402,30 +472,110 @@ namespace cubperf
     m_mmon = mmon;
   }
 
-  int memory_monitor::aggregater::get_server_info (const MMON_SERVER_INFO &info)
+  void memory_monitor::aggregater::get_server_info (MMON_SERVER_INFO &info) const
   {
-    return 0;
+    strncpy (info.name, m_mmon->m_server_name.c_str (), m_mmon->m_server_name.size ());
+    info.total_mem_usage = m_mmon->m_total_mem_usage.load ();
   }
 
-  int memory_monitor::aggregater::get_module_info (const MMON_MODULE_INFO &info, int module_index)
+  int memory_monitor::aggregater::get_module_info (MMON_MODULE_INFO &info, int module_index) const
```

Comment:
Nit. input-first seems better to me.
```suggestion
  int memory_monitor::aggregater::get_module_info (int module_index, MMON_MODULE_INFO &info) const
```

Reply:
@wrlawodms I agree with you if you mean input params followed by ouput params. @Rudeus Isn't the previous version correct in this case?